app/routes/achievements/components/Overview.tsx

```diff
@@ -0,0 +1,186 @@
+import { FilterSection, filterSidebar, Flex, Page } from '@webkom/lego-bricks';
+import { Trophy } from 'lucide-react';
+import { RadioButton } from 'app/components/Form';
+import AchievementsInfo, {
+  rarityToColorMap,
+} from 'app/utils/achievementConstants';
+import useQuery from 'app/utils/useQuery';
+import styles from './Overview.module.css';
+import { AchievementTabs } from './utils';
+
+const Overview = () => {
+  const { query, setQueryValue, setQuery } = useQuery({
+    min_rarity: 'any',
+    max_rarity: 'any',
+    sort: 'none',
+    sort_order: 'asc',
+  });
+
+  const allTrophies = Object.entries(AchievementsInfo).flatMap(
+    ([key, achievements]) =>
+      achievements.map((achievement, index) => ({
+        ...achievement,
+        identifier: key,
+        level: index,
+      })),
+  );
+
+  const filteredTrophies = allTrophies.filter((trophy) => {
+    const rarity = trophy.rarity;
+    const minRarity =
+      query.min_rarity !== 'any' ? parseInt(query.min_rarity, 10) : 0;
+    const maxRarity =
+      query.max_rarity !== 'any' ? parseInt(query.max_rarity, 10) : Infinity;
+    return rarity >= minRarity && rarity <= maxRarity;
+  });
+
+  const sortedTrophies = [...filteredTrophies].sort((a, b) => {
+    let comparison = 0;
+    if (query.sort === 'rarity') comparison = a.rarity - b.rarity;
+    if (query.sort === 'alphabetical')
+      comparison = a.name.localeCompare(b.name);
+    if (query.sort === 'hidden')
+      comparison = a.hidden === b.hidden ? 0 : a.hidden ? -1 : 1;
+
+    return query.sort_order === 'desc' ? -comparison : comparison;
+  });
```

Comment:
Is there a reason we are explicitly storing each list?
I get that it's nice to divide clearly up into filtering and sorting, but it might be better to just define the filter and sorting lambda's instead and then do

```js
const trophies = 
	Obejct.entries(Achievements)
	.flatMap(flatMapper)
	.filter(trophyFilter)
	.sort(trophyComparator)
```


where `flatMapper`, `trophyFilter`, and `trophyComparator` are just named lambdas:

```js
const trophyFilter: (trophy: Trophy) => boolean = (trophy)  => {
//...
} 
```

Reply:
An named lambda for the `flatMapper` isn't really that important though